### DISCUSSION TITLE
Why does my Neovim instance fail to compile to the parsers?

### DISCUSSION THREAD

**Discussion by jarmos-san** [nvim-treesitter/nvim-treesitter#1623] [Q&A]
I followed the official tutorials to install `nvim-treesitter` for Windows. Got MinGW toolsets installed, `gcc` is available on `$PATH`, everything appears to be fine, still Neovim fails to compile the parsers. I went through #1097 & #626 as well yet I still can't figure out what went wrong. So, any help to debug this issue further would be a godsend.

Besides, here's what `:CheckHealth nvim_treesitter` returns:

```console
 ## Installation
 - WARNING: `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
   - OK: `node` found v16.1.0 (only needed for :TSInstallFromGrammar)
   - OK: `git` executable found.
   - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl" }
   - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

 ## Parser/Features H L F I J

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
          +) multiple parsers found, only one will be used
          x) errors found in the query, try to run :TSUpdate {lang}
```

Wondering if I need to have the `tree-sitter` binary globally available on `$PATH`?

**Comment by theHamsta** [comment]
No Tree-Sitter binary is not needed for most parsers. If it is for a particular one you'll get notified.

What's the error message you're getting?

Have you tried the tips in the windows wiki (compiling with msvc or clang)?

**Reply by jarmos-san** [reply]
> Have you tried the tips in the windows wiki (compiling with msvc or clang)?

Yes I did. Tried compiling through LLVM (clang) & MSVC as well. Using `gcc` (MinGW) right now & they're available on `$PATH`, yet compiling fails.

This is the message I get trying to compile the Python parser:
```console
nvim-treesitter9python]: Error during compilation
```

Screenshot attached for further reference:

![image](https://user-images.githubusercontent.com/31373860/126872155-c7774233-10a7-4938-b6d5-9a7ec8a5753e.png)

**Reply by theHamsta** [reply]
Maybe we get the error output with TSInstallSync?

**Reply by jarmos-san** [reply]
> Maybe we get the error output with TSInstallSync?

Oh interestingly, `TSInstallSync python` says the parsers are downloaded & installed but `TSInstallInfo` has something else to show.

Screenshots attached for further reference:

Message from `TSInstallSync python`:

![image](https://user-images.githubusercontent.com/31373860/126875173-4c85a281-4fda-498d-88a7-c069de3f6d43.png)

Message from `TSInstallInfo`:

![image](https://user-images.githubusercontent.com/31373860/126875132-ff35d7ab-3605-48fe-bf9a-08ec545b8522.png)

**Reply by theHamsta** [reply]
Is your gcc visible as `cc` in your PATH? Which console are you using that could eat up all errors? Can you try to compile the parsers manually? The command should be `gcc -I./src src/parser src/scanner.cc -Os -shared -lstdc++ -o python.so`

**Reply by theHamsta** [reply]
This looks also suspicious.
```
    if fn.has "win32" == 0 then
      table.insert(args, "-fPIC")
    end
```

**Reply by jarmos-san** [reply]
> Is your gcc visible as `cc` in your PATH? Which console are you using that could eat up all errors? Can you try to compile the parsers manually? The command should be `gcc -I./src src/parser src/scanner.cc -Os -shared -lstdc++ -o python.so`

Yeah sure I can try that & I'm using Windows PowerShell (the v5.x one). And I have to get some work done so I downloaded the parsers from the CI & used them.

I don't mind compiling the parsers manually & it's on my TODO list right now, so do give me some time to get back to it.
